### PR TITLE
Adding a TCK test that clarifies values being empty as being present.

### LINKED ISSUE #216: Clarify config value presence
The API javadocs for `Config.getValue` mentions:

```
@throws java.util.NoSuchElementException if the property isn't present in the configuration.
```

Geronimo Config takes an over zealous approach to this, throwing NoSuchElementException if the property exists but has an empty value.  Payara takes a more conservative approach, only throwing this if the value is null.

The fact that the TCK doesn't test for empty string makes me thing the Payara thinking is more correct, but wanted to get confirmation (and possibly a TCK assertion for this).

### PR DESCRIPTION
**Pull request by johnament**
This closes #216

Signed-off-by: John D. Ament <johndament@apache.org>

### CODE REVIEW

File: tck/src/main/java/org/eclipse/microprofile/config/tck/EmptyValuesTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Test
+    public void testEmptyStringValues() {
+        assertEquals(emptyValuesBean.getStringValue(), "");
+    }
+
+    @Test
+    public void testEmptyStringProgrammaticLookup() {
+        System.setProperty(EMPTY_PROPERTY, "");
+        String stringValue = config.getValue(EMPTY_PROPERTY, String.class);
```
Comment (Emily-Jiang) [inline tck/src/main/java/org/eclipse/microprofile/config/tck/EmptyValuesTest.java]: I think putting the EMPTY_PROPERTY to the .properties is a better approach, as this test assume the system property will make to the runtime instantly.

Comment (johnament) [inline tck/src/main/java/org/eclipse/microprofile/config/tck/EmptyValuesTest.java]: @Emily-Jiang I'm not sure what you mean, this is the same approach used in `CDIPlainInjectionTest.ensure_all_property_values_are_defined`

Comment (Emily-Jiang) [inline tck/src/main/java/org/eclipse/microprofile/config/tck/EmptyValuesTest.java]: I prefer you add my.empty.property to microprofile-config.properties as it would represent the real use case. I think a lot of times to have them set blank and then override via a system property to set sometime more meaningful.

As the tck has the set the presence of using system property and expect it to appear as config straightaway. I am not going to block your commit though I think it can be approved. If @struberg or @OndrejM are fine with it, I am ok with it.

**Comment by Emily-Jiang** [comment]
@johnament Feel free to merge this one. We can improve it later if you feel differently.

File: tck/src/main/java/org/eclipse/microprofile/config/tck/EmptyValuesTest.java (hunk 1 above)
Comment (johnament) [inline tck/src/main/java/org/eclipse/microprofile/config/tck/EmptyValuesTest.java]: Ok, well now I understand what you're asking for.  I'll change it, makes sense.

Comment (johnament) [inline tck/src/main/java/org/eclipse/microprofile/config/tck/EmptyValuesTest.java]: Added additional test from property file.
